### PR TITLE
Add function to check event data for error

### PR DESCRIPTION
**Pull request by AbdiTolesa**
Fix https://github.com/Strategy11/formidable-pro/issues/4908

Reflects changes from https://github.com/Strategy11/formidable-stripe/pull/324 and https://github.com/Strategy11/formidable-stripe/pull/328 in lite.

## Summary by CodeRabbit

- **Bug Fixes**
    - Enhanced error handling in the payment confirmation process to include checks for a new type of error, improving reliability during form submissions.

### CODE REVIEW

**Comment by garretlaxton** [comment]
Is there something specific that should be tested here?

**Comment by garretlaxton** [comment]
When I test https://github.com/Strategy11/formidable-stripe/pull/324 I get an `Unable to find record of payment.` error.
image

I also see this `bool(false)` text above the submit button in the form builder.
image

**Comment by AbdiTolesa** [comment]
> When I test [Strategy11/formidable-stripe#324](https://github.com/Strategy11/formidable-stripe/pull/324) I get an `Unable to find record of payment.` error.

@garretlaxton I couldn't replicate this giving '0' in the amount field. Are there any specific steps I need to go through. Here is the output I get when I try that:
image

**Comment by garretlaxton** [comment]
@AbdiTolesa I just added a field with the default value of 0 and used the field ID for the amount.

**Review by garretlaxton — APPROVED** [review]
Thanks! This works great for me now!

File: stripe/controllers/FrmTransLiteActionsController.php [hunk 1]
```diff
@@ -85,9 +85,58 @@ public static function add_trigger_to_action( $options ) {
 	 */
 	public static function trigger_action( $action, $entry, $form ) {
 		self::prepare_description( $action, compact( 'entry', 'form' ) );
-		FrmStrpLiteActionsController::trigger_gateway( $action, $entry, $form );
+		$response = FrmStrpLiteActionsController::trigger_gateway( $action, $entry, $form );
+		if ( ! $response['success'] ) {
+			// the payment failed
+			if ( $response['show_errors'] ) {
```
Comment (truongwp) [inline stripe/controllers/FrmTransLiteActionsController.php:91]: We can merge these checks into one `if`.

File: stripe/controllers/FrmTransLiteActionsController.php [hunk 2]
```diff
@@ -85,9 +85,58 @@ public static function add_trigger_to_action( $options ) {
...
+				self::show_failed_message( compact( 'action', 'entry', 'form', 'response' ) );
+			}
+		}
+	}
+
+	/**
+	 * @param array $args
+	 * @return void
+	 */
+	public static function show_failed_message( $args ) {
```
Comment (truongwp) [inline stripe/controllers/FrmTransLiteActionsController.php:101]: Please add `@since` tag to all new public methods.

**Review by truongwp — APPROVED** [review]
This looks good. Thanks @AbdiTolesa!

File: stripe/controllers/FrmTransLiteActionsController.php [hunk 3]
```diff
@@ -85,9 +85,58 @@ public static function add_trigger_to_action( $options ) {
...
+		}
+	}
+
+	/**
+	 * @since x.x
+	 *
+	 * @param array $args
+	 * @return void
+	 */
+	public static function show_failed_message( $args ) {
```
Comment (Crabcyborg) [inline stripe/controllers/FrmTransLiteActionsController.php:101]: It looks like this function doesn't need to be `public`.

**Review by Crabcyborg — APPROVED** [review]
Thanks @AbdiTolesa @truongwp and @garretlaxton for working on this!

I think this is good to merge.

🚀

**Pull request merged**